### PR TITLE
JBTIS-382 - split JBTIS target platforms - release/early access

### PR DESCRIPTION
**Pull request by pleacu**
Split JBTIS TP along release/final and non-release/early access lines so as to not expose non .Final features to the stable capture that appear when performing an 'uncategorized' install.

### CODE REVIEW

File: devstudio/pom.xml [hunk 1]
```diff
@@ -49,7 +49,8 @@ http://download.jboss.org/jbosstools/builds/staging/JBDSIS-aggregate-disc/all/re
     <PREFIX>devstudio-integration-stack</PREFIX>
     <TARGET_PLATFORM>luna</TARGET_PLATFORM>
     <VERSION>8.0.0.CR2</VERSION>
-    <IS_TP_VERSION>4.2.0.Final</IS_TP_VERSION>
+    <IS_TP_VERSION>4.2.1.Final</IS_TP_VERSION>
+    <IS_TP_EA_VERSION>4.2.1.EA</IS_TP_EA_VERSION>
```
Comment (maxandersen) [inline devstudio/pom.xml]: the version is still 4.2.1.Final IMO

it is the artifact that changes, not the version.

File: devstudio/site-ea/pom.xml [hunk 2]
```diff
@@ -83,7 +83,7 @@
             <artifact>
               <groupId>org.jboss.tools.integration-stack</groupId>
               <artifactId>target-platform</artifactId>
```
Comment (maxandersen) [inline devstudio/site-ea/pom.xml]: earlyaccess-target-platform would be the proper name IMO

File: devstudio/site-ea/pom.xml [hunk 3]
```diff
@@ -83,7 +83,7 @@
...
-              <version>${IS_TP_VERSION}</version>
+              <version>${IS_TP_EA_VERSION}</version>
```
Comment (maxandersen) [inline devstudio/site-ea/pom.xml]: the version is still IS_TP_VERSION

File: devstudio/site/pom.xml [hunk 4]
```diff
@@ -138,7 +138,7 @@
             <artifact>
               <groupId>org.jboss.tools.integration-stack</groupId>
               <artifactId>target-platform</artifactId>
-              <version>${IS_TP_VERSION}</version>
+              <version>${IS_TP_EA_VERSION}</version>
```
Comment (maxandersen) [inline devstudio/site/pom.xml]: why is this Early access version when its the main site ?

Comment (maxandersen) [inline devstudio/site/pom.xml]: but again, its the arifact that changes not the version.

File: jbosstools/pom.xml [hunk 5]
```diff
@@ -42,14 +42,15 @@
...
+    <IS_TP_VERSION>4.2.1.Final</IS_TP_VERSION>
+    <IS_TP_EA_VERSION>4.2.1.EA</IS_TP_EA_VERSION>
     <tycho-version>0.21.0</tycho-version>
     <jboss-tycho-version>0.21.1</jboss-tycho-version>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <target.eclipse.version>4.4 (Luna) or &lt;a href="https://www.jboss.org/products/devstudio/overview/"&gt;JBoss Developer Studio 8&lt;/a&gt;</target.eclipse.version>
 
     <!-- The URL for the JBTIS target platform that should be associated with this site. -->
-    <targetplatform.url>http://download.jboss.org/jbosstools/builds/staging/JBTIS-target-platform/4.2.0.Final/REPO/</targetplatform.url>
+    <targetplatform.url>http://download.jboss.org/jbosstools/builds/staging/JBTIS-target-platform/${IS_TP_EA_VERSION}/REPO/</targetplatform.url>
```
Comment (maxandersen) [inline jbosstools/pom.xml]: shuold be something like:

JBTIS-earlyacces-target-platform/${IS_TP_VERSION}/

File: jbosstools/site/pom.xml [hunk 6]
```diff
@@ -167,7 +167,7 @@
             <artifact>
               <groupId>org.jboss.tools.integration-stack</groupId>
               <artifactId>target-platform</artifactId>
-              <version>${IS_TP_VERSION}</version>
+              <version>${IS_TP_EA_VERSION}</version>
```
Comment (maxandersen) [inline jbosstools/site/pom.xml]: same bug as above - this is no earlyaccess.

File: jbosstools/site-ea/pom.xml [hunk 7]
```diff
@@ -97,7 +97,7 @@
             <artifact>
               <groupId>org.jboss.tools.integration-stack</groupId>
               <artifactId>target-platform</artifactId>
-              <version>${IS_TP_VERSION}</version>
+              <version>${IS_TP_EA_VERSION}</version>
```
Comment (maxandersen) [inline jbosstools/site-ea/pom.xml]: same as above, change the artifact not the version

**Comment by maxandersen** [comment]
I only have limited time today but
main "bug" is that you are using the version to differentiate where IMO it is a *new* artifact that is being introduced, not a new new version.

Also even that was the case, 4.2.1.EA is *not* an allowed/valid version - so I suggest using it as a separate artifact instead with shared version since that is actually what is done here. split in Two TP's (or updatesites) for one version.

I also believe that would remove the need for all the profiling switching ? (but can't say that for sure since I haven't had
time to run the build to completeness)

**Comment by pleacu** [comment]
Supplanted by PR 383

**Pull request closed without merging**